app/models/competition.rb

```diff
@@ -1058,19 +1054,11 @@ def country_zones
     self.longitude_microdegrees = @longitude_degrees * 1e6 unless @longitude_degrees.nil?
   end
 
-  def has_events_with_ids?(event_ids)
-    (event_ids - events.ids).empty?
-  end
-
-  def has_event?(event)
-    self.events.include?(event)
-  end
-
-  def has_base_entry_fee?
+  def base_entry_fee_set?
```

Comment:
The method name suggests "the value is set", but it actually checks "the value is set and non-zero". What about `base_entry_fee_nonzero?`?

(which would probably end up becoming `delegate :nonzero?, to: :base_entry_fee, prefix: true` anyways)

Reply:
doesn't look like that works
```
Competition.find("WC2025").base_entry_fee_nonzero?
  Competition Load (0.5ms)  SELECT `Competitions`.* FROM `Competitions` WHERE `Competitions`.`id` = 'WC2025' LIMIT 1
=> #<Money fractional:12000 currency:USD>
```
But I am fine with the name